src/scripts/main.js

```diff
@@ -1,3 +1,41 @@
 'use strict';
 
-// write code here
+const listItems = document.querySelectorAll('li');
+
+function sortList(list) {
+  list.sort((person1, person2) => person2.salary - person1.salary);
+
+  list.forEach((employee, index) => {
+    const li = listItems[index];
+
+    li.textContent = employee.name;
+
+    // Обновляем атрибуты
+    li.setAttribute('data-position', employee.position);
+
+    li.setAttribute(
+      'data-salary',
+      `$${employee.salary.toLocaleString('en-US')}`,
+    );
+    li.setAttribute('data-age', employee.age);
+  });
+}
+
+function getEmployees(list) {
+  const employee = Array.from(list).map((person) => {
+    return {
+      name: person.textContent.trim(),
+      position: person.dataset.position,
+      salary: stringToNum(person.dataset.salary),
+      age: person.dataset.age,
+    };
+  });
+
+  return employee;
+}
+
+function stringToNum(str) {
+  return parseInt(str.replace(/[^0-9]/g, ''), 10);
+}
+
+sortList(getEmployees(listItems));
```

Comment:
The `sortList` function is being called with the result of `getEmployees(listItems)`, which returns an array of employee objects. Make sure that `sortList` is designed to handle this array correctly, as it currently expects an array of objects with `salary` properties.